apps/lib/s_socket.c

```diff
@@ -84,8 +84,9 @@ int init_client(int *sock, const char *host, const char *port,
     int ret;
     int options = 0;
 
-    if (tfo && ba_ret != NULL)
-        *ba_ret = NULL;
+    /* There might be a BIO_ADDR from previous calls, release it. */
+    BIO_ADDR_free(*ba_ret);
+    *ba_ret = NULL;
```

Comment:
Hmm, this contradicts the doc comment above, which says "@ba_ret: ... to be freed by caller".

It would seem reasonable for me to use this function without initialising `*ba_ret` based on that documentation, e.g.

```
BIO_ADDR *peer_addr;
:
int ret = init_client(&sock, host, port, bindhost, bindport, socket_family,                                                           
                    socket_type, protocol, tfo, !isquic, &peer_addr);
:
```

This change would break such a usage.

`peer_addr` doesn't appear to be documented as in/out, which might mitigate this problem.

If `init_client()` is purely an internal function to OpenSSL this is okay though - can you confirm that?

Reply:
Of course init_client() is purely internal. The argument is in/out. Let me do this differently though.

Reply:
Sorry, just wanted to check